src/main/kotlin/org/rust/ide/docs/RustDocumentationProvider.kt

```diff
@@ -60,15 +60,35 @@ class RustDocumentationProvider : AbstractDocumentationProvider() {
         }
 
         val signatureStart = signatureStartElement?.startOffsetInParent ?: 0
-        val signatureEnd = block?.startOffsetInParent ?: textLength
+
+        // pick last of: where clause, return type, or closing brace of the parameters
+        // if all else fails, drop down to the length of the current element
+        val signatureEnd = if (block != null)
+            block!!.startOffsetInParent
+        else if (whereClause != null)
+            whereClause!!.let { it.startOffsetInParent + it.textLength }
+        else if (retType != null)
+            retType!!.let { it.startOffsetInParent + it.textLength }
+        else if (parameters != null)
+            parameters!!.let { it.startOffsetInParent + it.textLength }
+        else
```

Comment:
Unfortunately `firstNotNull` doesn't exist... Is there some import I have to have?